pkg/scaffold/resource/resource.go

```diff
@@ -73,9 +73,17 @@ func (r *Resource) Validate() error {
 		return fmt.Errorf(
 			"version must match ^v\\d+(alpha\\d+|beta\\d+)?$ (was %s)", r.Version)
 	}
-	// Check if the Kind is a valid value
-	if r.Kind != flect.Pascalize(r.Kind) {
-		return fmt.Errorf("kind must be PascalCase (expected %s was %s)", flect.Pascalize(r.Kind), r.Kind)
+
+	validationErrors := []string{}
```

Comment:
Just a minor comment? Could this be a `[]error{}`?

Reply:
agreed with @Adirio  here -- make the the corresponding function return `[]error`

Reply:
Sure 👍 However, a lot of other functions in that file use the same pattern (collect error messages in a  `[]string`, return an error in the end with all the error messages)

e.g. (from current master): https://github.com/kubernetes-sigs/kubebuilder/blob/cf6cec42bdf594eac67cfb8f13eee6f57ba5c47b/pkg/scaffold/resource/resource.go#L126-L135

It traces back to `func maxLenError(length int) string {` returning a string, so it would be a broader change. It's pretty straightforward, just checking that it would be fine to include in this PR.

Reply:
(treats me right for saying `It's pretty straightforward` :) ) This change would also require refactoring test expectations slightly. 

e.g.  `Expect(err).To(MatchError(ContainSubstring("a DNS-1035 label must consist of lower case alphanumeric characters")))` 
would fail if err now holds a slice of `error`s rather than `string`s. Again, happy to go ahead with it, just making sure it would make sense.

Reply:
oh, if the rest of the file does something similar, it's fine to leave for now :-).  Up to you.  Let me know what you decide.

Reply:
Left it as it was for this PR in the end. I will spend a bit of time right after to experiment, and will open a small cleanup PR if I figure out a way to work with `error`s in `resource.go` while keeping the test cases readable enough.